docs/cookbook/publishing_with_workflow.md

```diff
@@ -0,0 +1,411 @@
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Transition;
+
+class PublishOnLastStageSubscriber implements EventSubscriberInterface
+{
+    /** @var \eZ\Publish\API\Repository\PermissionResolver */
+    private $permissionResolver;
+
+    /** @var \EzSystems\EzPlatformWorkflow\Registry\WorkflowDefinitionMetadataRegistry */
+    private $workflowMetadataRegistry;
+
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /** @var \EzSystems\FlexWorkflow\API\Repository\RepositoryInterface */
+    private $repository;
+
+    private $publishOnLastStageWorkflows;
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \eZ\Publish\API\Repository\Repository $repository
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     * @param \EzSystems\EzPlatformWorkflow\Registry\WorkflowDefinitionMetadataRegistry $workflowMetadataRegistry
+     * @param array $publishOnLastStageWorkflows
+     */
+    public function __construct(
+        PermissionResolver $permissionResolver,
+        Repository $repository,
+        ContentService $contentService,
+        WorkflowDefinitionMetadataRegistry $workflowMetadataRegistry,
+        array $publishOnLastStageWorkflows
+    ) {
+        $this->permissionResolver = $permissionResolver;
+        $this->workflowMetadataRegistry = $workflowMetadataRegistry;
+        $this->contentService = $contentService;
+        $this->repository = $repository;
+        $this->publishOnLastStageWorkflows = $publishOnLastStageWorkflows;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            WorkflowEvents::WORKFLOW_STAGE_CHANGE => ['onStageChange', 30],
+        ];
+    }
+
+    public function onStageChange(StageChangeEvent $event): void
+    {
+        $workflowName = $event->getWorkflowMetadata()->name;
+
+        if (!\in_array($workflowName, $this->publishOnLastStageWorkflows, true)
+            || !$this->workflowMetadataRegistry->hasWorkflowMetadata($workflowName)
+        ) {
+            return;
+        }
+
+        $workflowDefinitionMetadata = $this->workflowMetadataRegistry->getWorkflowMetadata($workflowName);
+        $workflowDefinition = $event->getWorkflowMetadata()->workflow->getDefinition();
+
+        $transitionName = $event->getTransitionMetadata()->name;
+        $workflowTos = $this->getWorkflowTransitionTos($transitionName, $workflowDefinition);
+
+        foreach ($workflowTos as $stageName) {
+            $isLastStage = $workflowDefinitionMetadata->getStageMetadata($stageName)->isLastStage();
+
+            if ($isLastStage) {
+                $this->permissionResolver->sudo(function () use ($event) {
+                    if ($event->getWorkflowMetadata()->versionInfo->isDraft()) {
```

Comment:
I'd add a comment here that it uses `sudo()` to ignore user permissions and this is risky. We should rather encourage people to set up permissions properly. In this case it would be to add `content:publish` policy on last workflow stage.